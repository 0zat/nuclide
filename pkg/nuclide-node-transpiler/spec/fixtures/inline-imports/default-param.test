import Foo from './foo';

class Bar {
  baz(n = Foo()) {}
}
