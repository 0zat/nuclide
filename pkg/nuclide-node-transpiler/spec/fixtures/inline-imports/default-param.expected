var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _foo;

function _load_foo() {
  return _foo = _interopRequireDefault(require('./foo'));
}

var Bar = (function () {
  function Bar() {
    _classCallCheck(this, Bar);
  }

  _createClass(Bar, [{
    key: 'baz',
    value: function baz() {
      var n = arguments.length <= 0 || arguments[0] === undefined ? (0, (_foo || _load_foo()).default)() : arguments[0];
    }
  }]);

  return Bar;
})();
